File: Oyas/bloom.py
import hashlib
class BloomFilter:
    def __init__(self, array_size, number_of_hashes):
        self.size = array_size
        self.hash_count = number_of_hashes
        self.bits = [0] * array_size

    def add_item(self, item):
       
        for i in range(self.hash_count):
            hash_result = hashlib.md5(f"{i}{item}".encode()).hexdigest()
            index = int(hash_result, 16) % self.size
            
            self.bits[index] = 1
           

    def check_item(self, item):
        for i in range(self.hash_count):
            hash_result = hashlib.md5(f"{i}{item}".encode()).hexdigest()
            index = int(hash_result, 16) % self.size
            
            if self.bits[index] == 0:
                return False 
        
        return True

my_filter = BloomFilter(20, 3)

my_filter.add_item("Cat")
my_filter.add_item("Dog")

print(f"Is 'Cat' in there? {my_filter.check_item('Cat')}")

print(f"Is 'Bird' in there? {my_filter.check_item('Bird')}")

print(f"Final bit array: {my_filter.bits}")

File: LakshmyKanth/bloom.c
#include <stdio.h>
#include <string.h>

#define SIZE 20     
#define HASH_COUNT 3 
int bitArray[SIZE] = {0};

int hashFunction(char *item, int seed) {
    int hash = 0;

    for (int i = 0; item[i] != '\0'; i++) {
        hash = hash + item[i] + seed;
    }

    return hash % SIZE;
}

void add(char *item) {
    for (int i = 0; i < HASH_COUNT; i++) {
        int index = hashFunction(item, i);
        bitArray[index] = 1; 
    }
}

int contains(char *item) {
    for (int i = 0; i < HASH_COUNT; i++) {
        int index = hashFunction(item, i);
        if (bitArray[index] == 0) {
            
            return 0;
        }
    }
    
    return 1;
}

int main() {
    add("Cat");
    add("Dog");

    if (contains("Cat"))
        printf("Cat is present (maybe)\n");
    else
        printf("Cat is not present\n");

    if (contains("Bird"))
        printf("Bird is present (maybe)\n");
    else
        printf("Bird is not present\n");

    return 0;
}